src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -236,9 +245,9 @@ void drawing_classifier::init_options(
 
   options.create_string_option(
       "warm_start",
-      "Record warm start model version used. If no warmstart used"
-      " None is assigned by default."
-      "",
+      "Record warm start model version used. If no warmstart used, "
+      "empty string is assigned by default.",
+      FLEX_UNDEFINED,
```

Comment:
later you cannot assign `UNDEFINED` for key `warm_start` through `add_or_update_state` because it expects a string type input, not `None`.

Reply:
I think if you cannot reset the value of the key to its default value, then what's the purpose of setting that unreachable value as default?

Reply:
I see other instances in the codebase where we assigned `FLEX_UNDEFINED` into the model state, e.g. in the `supervised_learning` toolkits. Maybe you just can't do it via `add_or_update_state`, but that may just be a flaw of that implementation.

Anyway, what's more important is the behavior the user sees. I think using `FLEX_UNDEFINED` here is more likely to give us robust behavior here, insofar as this value translating into what the user expects if they query the `warm_start` property on the model class.